backends/transforms/utils.py

```diff
@@ -187,3 +230,128 @@ def delete_constant_placeholder(exp_program: ExportedProgram, node: torch.fx.Nod
 
     # Remove node from graph
     node.graph.erase_node(node)
+
+
+def create_mutable_buffer(
+    exp_program: ExportedProgram,
+    graph: torch.fx.Graph,
+    name: str,
+    data: torch.Tensor,
+) -> torch.fx.Node:
+    """
+    Creates and returns a mutable buffer placeholder node. This is similar to
+    create_constant_placeholder but specifically for creating mutable buffers that
+    can be modified during execution.
+
+    The difference between this and create_constant_placeholder is that this doesn't
+    expect user to set the correct position for the placeholder node to be inserted,
+    it finds the correct position automatically.
+
+    Args:
+        exp_program: The exported program to modify
+        graph: The graph to add the node to
+        name: The name for the new buffer node (should start with "b_" prefix by convention)
+        data: The initial tensor data for the buffer
+
+    Returns:
+        The created placeholder node to be used in the graph
+    """
+    # Input validation
+    if not name or not name.strip():
+        raise ValueError("Buffer name cannot be empty")
+
+    if not isinstance(data, torch.Tensor):
+        raise ValueError("Data must be a torch.Tensor")
+
+    # Extract target name (remove "b_" prefix if present, following export convention)
+    if name.startswith("b_"):
+        target = name[2:]
+    else:
+        target = name
+
+    # Check if target already exists
+    if target in exp_program.state_dict:
+        raise RuntimeError(f"Buffer target '{target}' already exists in state_dict")
+
+    persistent_buffer = True
+    exp_program.state_dict[target] = data
+
+    # Create fake tensor using helper function
+    fake_tensor = _get_fake_tensor_mode(graph, data)
+
+    # Signature ordering is as follows:
+    # Inputs = [*parameters_buffers_constant_tensors, *flattened_user_inputs]
+    #                       ^^^^^^^
+    # Outputs = [*mutated_inputs, *flattened_user_outputs]
+
+    # Validate that the signature is up to date and aligned with the placeholders in the graph
+    placeholders = [n for n in graph.nodes if n.op == "placeholder"]
+    if len(placeholders) != len(exp_program.graph_signature.input_specs):
+        raise RuntimeError(
+            f"Graph has {len(placeholders)} placeholder nodes but signature has "
+            f"{len(exp_program.graph_signature.input_specs)} input specs"
+        )
+
+    for node, input_spec in zip(placeholders, exp_program.graph_signature.input_specs):
+        if node.name != input_spec.arg.name:
+            raise RuntimeError(
+                f"Input node {node.name} does not match input spec {input_spec.arg.name}"
+            )
+
+    # Find indices of user inputs and constant tensors
+    user_input_indices_or_consts = [
```

Comment:
I feel like a loop would be a cleaner way of writing this through line 324. 

Since you have validated the Spec.kind and the nodes earlier you could just loop over the input_specs and index into graph.nodes